client/src/components/PacmanGame/gameCore.js

```diff
@@ -117,18 +117,27 @@ export const entityToCode = (entity) => {
 
 export const isWall = (gridState, { x, y }) => Boolean(gridState[x][y] === entityToCode('wall'));
 
-const choices = [
-  [0, -1, 'TOP'],
-  [0, 1, 'DOWN'],
-  [1, 0, 'RIGHT'],
-  [-1, 0, 'LEFT'],
-];
+const choices = {
+  UP: [0, -1],
+  DOWN: [0, 1],
+  RIGHT: [1, 0],
+  LEFT: [-1, 0],
+};
 
 export const getRandomAdjacentAvailableCell = (gridState, currentLocation) => {
-  const randomIndexInChoice = parseInt(Math.random() * choices.length, 10);
+  const { x, y, direction } = currentLocation;
+  const listOfChoices = Object.keys(choices);
+  const randomIndexInChoice = parseInt(Math.random() * listOfChoices.length, 10);
+  const randomChoice = choices[listOfChoices[randomIndexInChoice]];
+  const totalX = choices[direction][0] + randomChoice[0];
```

Comment:
If using `x,y` property in `choices`, this can be converted to more readable form of `choices[direction][0].x` and `randomChoice.x` instead of using 0 and 1.